mux.go

```diff
@@ -11,15 +11,22 @@ import (
 	"net/http"
 	"path"
 	"regexp"
+	"strings"
 )
 
-var (
-	// ErrMethodMismatch is returned when the method in the request does not match
-	// the method defined against the route.
-	ErrMethodMismatch = errors.New("method is not allowed")
-	// ErrNotFound is returned when no route match is found.
-	ErrNotFound = errors.New("no matching route was found")
-)
+// ErrMethodMismatch is returned when the method in the request does not
+// match the method defined against the route.
+type ErrMethodMismatch struct {
+	// The methods that the matcher would have allowed.
+	AllowedMethods []string
+}
+
+func (ErrMethodMismatch) Error() string {
+	return "method is not allowed"
+}
```

Comment:
Changing `ErrMethodMismatch` this way is not backwards compatible. Existing user code checking for `err == ErrMethodMismatch` will break after this change because `ErrMethodMismatch` is not a value anymore.

Reply:
You’re right, and I agree with your suggestion. Will change the PR accordingly.